node/src/blockchain/blockchain_bridge.rs

```diff
@@ -232,229 +246,238 @@ impl BlockchainBridge {
+                    for transaction_receipt in &transaction_receipts_results {
+                        if let TransactionReceiptResult::Found(_) = transaction_receipt {
+                            transactions_found += 1;
+                        }
+                    }
+                    let pairs = transaction_receipts_results
+                        .into_iter()
+                        .zip(msg.pending_payable.into_iter())
+                        .collect_vec();
+                    accountant_recipient
+                        .try_send(ReportTransactionReceipts {
+                            fingerprints_with_receipts: pairs,
+                            response_skeleton_opt: msg.response_skeleton_opt,
+                        })
+                        .expect("Accountant is dead");
+                    if length != transactions_found {
+                        return Err(format!(
+                            "Aborting scanning; {} transactions succeed and {} transactions failed",
+                            transactions_found,
+                            length - transactions_found
+                        ));
+                    };
+                    Ok(())
+                }),
+        )
     }
 
-    fn handle_scan<M, F>(&mut self, handler: F, scan_type: ScanType, msg: M)
+    fn handle_scan_future<M, F>(&mut self, handler: F, scan_type: ScanType, msg: M)
     where
-        F: FnOnce(&mut BlockchainBridge, M) -> Result<(), String>,
+        F: FnOnce(&mut BlockchainBridge, M) -> Box<dyn Future<Item = (), Error = String>>,
         M: SkeletonOptHolder,
     {
         let skeleton_opt = msg.skeleton_opt();
-        match handler(self, msg) {
-            Ok(_r) => (),
-            Err(e) => {
-                warning!(self.logger, "{}", e);
-                self.scan_error_subs_opt
-                    .as_ref()
-                    .expect("Accountant not bound")
-                    .try_send(ScanError {
-                        scan_type,
-                        response_skeleton_opt: skeleton_opt,
-                        msg: e,
-                    })
-                    .expect("Accountant is dead");
-            }
-        }
+        let logger = self.logger.clone();
+        let scan_error_subs_opt = self.scan_error_subs_opt.clone();
+        let future = handler(self, msg).map_err(move |e| {
+            warning!(logger, "{}", e);
+            // TODO: This ScanError needs to be removed, And added into OutboundPaymentsInstructions & QualifiedPayablesMessage
+            scan_error_subs_opt
+                .as_ref()
+                .expect("Accountant not bound")
+                .try_send(ScanError {
+                    scan_type,
+                    response_skeleton_opt: skeleton_opt,
+                    msg: e,
+                })
+                .expect("Accountant is dead");
+        });
+
+        actix::spawn(future);
     }
 
     fn process_payments(
```

Comment:
I think it would nice if this method could a more confusion-avoiding name. Maybe "obey_instructions_for_payments", "generate_payments", "pay_to_creditors",...

Reply:
I'd really appreciate to have this.